app/controllers/api/v1/teachers_controller.rb

```diff
@@ -103,6 +103,71 @@ def name_valid
     end
   end
 
+  def get_enews_subscription
+    teacher_id = params.require(:id)
+    email = User.find(teacher_id).email
+
+    @mc_api_key = ENV['MAILCHIMP_API_KEY']
+    @mc_list_id = ENV['MAILCHIMP_API_LISTID']
+    @mc_uri = ENV['MAILCHIMP_API_URI']
+    @mc_data = {
+      'email_address' => "#{email}"
+      }
+    @digest = Digest::MD5.hexdigest("#{email}")
+
+    uri = URI("#{@mc_uri}/#{@mc_list_id}/members/#{@digest}")
+    http = Net::HTTP.new(uri.host, uri.port)
+    http.use_ssl = true
+    req = Net::HTTP::Get.new(uri.path, 'Content-type' => 'application/json')
+    req.basic_auth("user", "#{@mc_api_key}")
+    req.body = @mc_data.to_json
+    response = http.request(req)
+    response_data = JSON.parse(response.body)
+    status = response_data['status']
+
+    if status == 'subscribed'
+      subscribed = true
+    else
+      subscribed = false
+    end
+
+    return render :json => {'params_id' => "#{params.require(:id)}", 'id' => "#{teacher_id}", 'status' => "#{status}", 'email_address' => "#{email}", 'subscribed' => "#{subscribed}"}
+  end
+
+  def update_enews_subscription
+    teacher_id = params.require(:id)
```

Comment:
Since this is on the teachers controller it would be better to pass an id for a Portal::Teacher object here.